### ISSUE TITLE
Evaluate different settings for growing training set

### ISSUE THREAD

**Issue by schrum2** [schrum2/MarioDiffusion#157]
Now that #117 is closed, we want to figure out how much of a benefit it really provides in a fairly systematic way, so I want you to create some batch files for running experiments as well as some minor code changes.

Specifically, in the batch directory make a file like train-conditional.bat that is named train-conditional-growing.bat. It should be a copy of train-conditional.bat with the following changes:

Remove the <game> and <type> parameters. Hard-code <game> to be Mar1and2 and <type> to be regular. This will simplify the script a bit. Keep the <seed> parameter first.

The script will start by training an MLM model in the same way, but the train_diffusion call will be different. It should always use --auto_augment and should also set various auto_augment parameters. The specific settings should depend on command line parameters sent to the batch script, and should affect the following arguments:

--auto_augment_threshold
--auto_augment_max_new_samples
--auto_augment_max_dataset_size

The values assigned to these parameters should affect the names of model output directories. For example, with a threshold of 0.8, max new samples of 10, and max dataset size of 10000, the diffusion model should be saved to something like:

Mar1and2-conditional-regular-thres0.8new10max10000-0

That final 0 at the end is from a <seed> value of 0, but this would be different with different seeds. The calls to run diffusion and evaluate caption adherence also need to be aware of this different directory name, so that they process results on the correct diffusion model.

Having this batch file will make it very easy to run a lot of experiments with different settings. We can launch a single command that trains and evaluates models, allowing us to compare which settings work the best. However, there is one more important change needed, and this needs to be in the code itself.

If we want to compare how well models do with training sets that grow in different ways, then we need a fair way to compare them. What if growing the dataset more aggressively leads to better results, but also takes way longer to train? This seems likely, since more data means longer training times. This isn't a fair comparison. Currently, the code is configured to run for a certain number of epochs, and it also plots performance based on the number of epochs. However, to do a fair comparison here, we need to measure performance based on the number of batches that the model is trained on. I think that we can still run the code for a fixed number of epochs, as long as we know how many iterations it took to reach each checkpoint, so this requires a change to the logging. It looks like this is already tracked via the global_step variable which appears in most jsonl log files as "step". So:

- add "step" to the dataset size log file as well.
- currently, each of these log files plots according to the epoch on the x-axis. Make it so that two plot files are produced for each jsonl log: one based on epoch (like what we already have) and one based on "step".

**Comment by almzayena-lab** [comment]
<img width="1919" height="1079" alt="Image" src="https://github.com/user-attachments/assets/14bdaa20-d642-4b59-a410-0dd4d94c6e8a" />

Thus far this is where I'm at, unfortunately its looking like the graphing is identical for both step and epoch so its likely simply changing the text rather then actually accumulating good data.

**Comment by schrum2** [comment]
I think the two plots are different, but just look very similar. I made my own test with some data I have and got this:

<img width="2257" height="1070" alt="Image" src="https://github.com/user-attachments/assets/0459ce38-ef6d-4ddc-9d01-fcd0fe4c8288" />

The shape of the curves is pretty much the same, but the vertical line intervals do not match, indicating different time scales. The code I used to make this is here:

[stepvsepoch.py](https://github.com/user-attachments/files/28578073/stepvsepoch.py)

For now, let's just assume the steps are being tracked correctly, and focus on the other issues.

I do have one more additional request though. Let's add a command line parameter that specifies the max iterations. At the end of each epoch, compare the global step to this value, and if the global_step is greater, then exit the loop and end training. Make the default value for this parameter be infinity (`float("inf")`)

**Comment by almzayena-lab** [comment]
<img width="834" height="426" alt="Image" src="https://github.com/user-attachments/assets/016dc541-1358-417d-bf68-8e871f61f55c" />

This is the progress thus far, i am going to leave it to run with max iteration of 100,000 to ensure it truly works.

**Comment by almzayena-lab** [comment]
<img width="948" height="349" alt="Image" src="https://github.com/user-attachments/assets/645cc96f-3995-4cbe-be5d-cea80ac1f44d" />

I have officially created a trained model which is based off of step amount rather then epoch.
